skopt/space/space.py

```diff
@@ -97,6 +97,8 @@ def check_dimension(dimension, transform=None):
 class Dimension(object):
     """Base class for search space dimensions."""
 
+    prior = "uniform"
```

Comment:
any reason behind hard coding this?

Reply:
That's a good questions: the problem is a missing implimentation of `Integer.prior`.  Now, I saw a couple of options:

* Impliment a stub in Integer.__init__.  The downside is that a new inherited class of `Dimension` would have the same problem if it doesn't impliment `prior`.

* Create a if/else in the plots.py code i added.  To my preference, this seemed ugly.

* Add `prior` to the base class (which is what I did in a minimally invasive way).  This could also be done via `super().__init__`, but would add a lot more code and inherited classes should remember to call `super().__init__` (see problem 1).

i the current solution is not good, but the best solution for this particular patch.

Reply:
This is a bit tricky. I would probably prefer a default value of `None`, but this means we need a check in the plotting ... the reason to prefer `None` is that `prior` doesn't have to be a string, for example in `space.Categorical`. Not sure what the best options is right now.

Reply:
sure, we could do `None`.  I believe though that `uniform` is more appropriate because assuming *no* prior just means that you accept a uniform prior across the specified interval/indices/categories.